docs/DeveloperGuide.md

```diff
@@ -6,7 +6,154 @@
+  processing on it:
+  ```java
+    @Override
+    protected void loadData() throws Exception {
+        this.dishes = new ArrayList<>();
+        ArrayList<?> list = (ArrayList<?>) this.load();
+        for (Object object : list) {
+            this.addDish((Dish) object);
+        }
+    }
+  ```
+- `public abstract void saveData() throws Exception`
+  - This is a convenience method for doing any pre- / post-processing on any data you may want to save. Of course, you should
+  also call the `save(Object o)` method finally.
+
+Note that loading and saving is only done once. Loading is done once upon programme start-up, and saving is done
+when the user exits the menu gracefully.
+
+Here is the typical design of a manager:
+
+![](diagrams/class/manager.png)
+
+
+### Menu (dishes) Management
+
+The management of menu is handled by `DishManager`, which implements following methods:
+
+- `void printDishes()` -- print all dishes and their menu index in current menu.
+- `void addDish(Dish dish)` -- add a new dish.
+- `void deleteDish(int menuIndex)` -- delete a dish using its menu index.
+- `void setPrice(int menuIndex, double newPrice)` -- change the price of a dish using its menu index.
+- `void setName(int menuIndex, String newName)` -- change the name of a dish using its menu index.
+
+These functions are "delegated" to `ArrayList<Dish>`, which is associated with `DishManager`.
+
+For example, `printDishes()`, iterates through `dishes` and calls `get(int)` method (`setPrice(int, double)`, `setName(int, String)` also call `get(int)`):
+
+![](diagrams/sqeuence/DishManager_printDishes.png)
+
+`deleteDish(int)` calls `remove(int)`:
+
+![](diagrams/sqeuence/DishManager_deleteDish.png)
+
+`addDish(Dish)` calls `add(Dish)`:
+
+![](diagrams/sqeuence/DishManager_addDish.png)
+
+
+
+### Order Management
+
+The management of menu is functioned by `OrderManager`, which implements following methods:
+- `void addDishToOrder(Dish dish, int orderIdx)` -- add dishes in to an order.
+- `void deleteOrder(int orderIndex)` -- delete an order using its order index.
+- `Order getOrder(int orderIndex)` -- get an order using its order index.
+- `double getOrderPrice(int orderIndex)` -- get the total price of an order using its order index.
+- `double getAllOrderValue()` -- get the total price of all orders.
+
+These functions are "delegated" to `ArrayList<Order>`, which is associated with `OrderManager`. 
+
+Methods call `get(int)` to get the element.
+
+For example, `addDishToOrder(Dish dish, int orderIdx)`, iterates through `orders` and calls `addDish(Dish dish)` method:
+
+![](diagrams/sqeuence/OrderManager_addDishToOrder.png)
+
+`deleteOrder(int)` calls `getOrderCount()`:
+![](diagrams/sqeuence/OrderManager_deleteOrder.png)
+
+Both methods `getOrderPrice(int)` and `getAllOrderValue()` call `getTotalPrice()`:
+![](diagrams/sqeuence/OrderManager_displayAllOrderPrice.png)
```

Comment:
<img width="238" alt="Screen Shot 2022-04-01 at 10 22 42 AM" src="https://user-images.githubusercontent.com/73009417/161182083-e6897e52-c757-48e8-80e5-b405760e4ee1.png">
Is it better to have a shorter activation bar for **Order** object?